specs/typography/heading.md

```diff
@@ -0,0 +1,241 @@
+
+Bare-bones styles applied manually.
+
+```TypeScript
+headingStyles.basicRamp()
+```
+
+Produces css classes that can be applied to any text elements. The classes are consistently named for relation to the raw type ramp.
+
+*Supports use cases 1 & 3, feature 1*
+
+```CSS
+.fast-heading-minus-2 ...
+
+.fast-heading-minus-1 ...
+
+.fast-heading-base {
+    font-size: var(--fast-type-ramp-base-size);
+    line-height: var(--fast-type-ramp-base-height);
+    font-weight: var(--fast-heading-font-weight);
+}
+
+.fast-heading-plus-1 ...
+
+...
+```
+
+Usage:
+
+```HTML
+<h1 class="fast-heading-plus-3">If all the world is a stage, I want better lighting</h1>
+
+<span class="fast-heading-minus-2">If all the world is a stage, I want better lighting</span>
+```
+
+![](./images/basic-heading-samples.png)
+
+#### Scoped standard heading element map
+
+The easiest option for accessible content when you know how many levels you have.
+
+```TypeScript
+headingStyles.standardElementMap(levels: 3, includeBase: false)
+```
+
+Applies the styles to the standard heading elements for the desired number of levels so the adaptive styling is used without needing to add and maintain class names. Starts directly above the base type ramp size or optionally includes that size depending on stylistic preference.
+
+*Supports use case 2, features 2 & 3*
+
+```CSS
+h1, h2, h3, h4, h5, h6 {
+    font-weight: var(--fast-heading-font-weight);
+}
+
+body {
+    font-size: var(--fast-type-ramp-base-size);
+    line-height: var(--fast-type-ramp-base-height);
+}
+
+/* Three levels, above base */
+
+h1 {
+    font-size: var(--fast-type-ramp-plus-3-size);
+    line-height: var(--fast-type-ramp-plus-3-height);
+}
+
+h2 {
+    font-size: var(--fast-type-ramp-plus-2-size);
+    line-height: var(--fast-type-ramp-plus-2-height);
+}
+
+h3 {
+    font-size: var(--fast-type-ramp-plus-1-size);
+    line-height: var(--fast-type-ramp-plus-1-height);
+}
+
+h4, h5, h6 { /* This rule may not be necessary, but it helps identify usage or config issues. */
+    font-size: var(--fast-type-ramp-base-size);
+    line-height: var(--fast-type-ramp-base-height);
+}
+```
+
+Usage:
+
+```HTML
+<h1>If all the world is a stage, I want better lighting</h1>
+
+<p>Studies show the quality of light affects people in many different ways. For example, productivity can be positively affected by well-designed illumination.</p>
+
+<h2>Human Needs</h2>
+
+<p>It's no secret that people are attracted to well-lighted public facilities, commercial shopping districts and parks.</p>
+
+<h3>Mood and Atmosphere</h3>
+
+<p>Good lighting enhances the mood and desirability of these spaces. It contributes greatly to people's sense of well-being.</p>
+```
+
+![](./images/standard-map-sample.png)
+
+#### Semantically named use case based styles
+
+An option for applying names to common scenarios. Serves as a common example as a guide for how to create your own semantic styles.
+
+```TypeScript
+headingStyles.simplePageStyleMap()
```

Comment:
I think what we'll want to do for these style-sets is expose them as:
1. Individual stylesheets that can be composed into larger stylesheets
2. Expose the "set" as a stylesheet that can be exposed globally or through a root control

Reply:
I was reconsidering the underlying approach here. Based on some comment from Chris, I think the idea of mapping a ramp value to a use case name should exist at the general type ramp level. Like a function or whatever that would take name / value pairs where like:
page-title: 3,
section-title: 2,
item-title: 1,
caption: -1,
random-large-text: 6

The heading example here would use that and essentially be an example of usage, or could be brought in as you mention.